### ISSUE TITLE
Build is broken in Node 6 due to Chrome rejecting the SSL server's certificate

### ISSUE THREAD

**Issue by m90** [mantoni/mochify.js#171]
So I [fixed a typo in the README](https://github.com/mantoni/mochify.js/commit/ff62e65b3e956d0c1f0f8ad6196a1df2754ab39e) and noticed that the build is broken in Node 6 since [recently](https://github.com/mantoni/mochify.js/commit/2ac6322362712bf563fcbd527733a12213932065), most likely unrelated to the changeset (all changes since the last passing build in Node 6 have been README only).

It seems to reproducibly fail in Node 6 now due to errors like:

```
Uncaught AssertionError: 'This site does not have a valid SSL certificate! Without SSL, your site\'s and visitors\' data is vulnerable to theft and tampe == '# chromium:\n1..1\nlocation.href = https://localhost:8080/test.html\nok 1 url has H1 element\n# tests 1\n# pass 1\n# fail 0\n'
```

I don't really understand what makes Chrome reject the cert when using Node 6, when Node 8 passes, but this is what I found out until now:

- test will pass just fine for me locally running Node `v8.11.1`
- the [last passed build](https://travis-ci.org/mantoni/mochify.js/jobs/374972795) on Node 6 was apparently using the very same Node/npm versions as the [failing ones](https://travis-ci.org/mantoni/mochify.js/jobs/377430935), so it's likely a Chrome / Pupeteer change:

```
$ node --version
v6.14.2
$ npm --version
3.10.10
$ nvm --version
0.33.11
```

Interestingly passing `--ignore-ssl-errors` to the tests [does **not** silence the failures](https://travis-ci.org/mantoni/mochify.js/builds/377442545) (only 98% sure I am [doing this](https://github.com/mantoni/mochify.js/commit/e0d41e776581e8bbaf9166d63370759120841d7a) correctly though).

Unfortunately I don't have a Node 6 test suite of a real project up and running so I could test if this is something that requires action as it breaks the real world or if it's just a weird Travis glitch, but it's probably a good idea to keep an eye on it and try to drill down to what the real cause of this is.

**Comment by m90** [comment]
I also [tried running this against more Node versions](https://travis-ci.org/mantoni/mochify.js/builds/377448022) and it looks like it's failing lt 8.

**Comment by mantoni** [comment]
Thanks for looking into it. I tested locally with a fresh Chromium download and it worked fine on Node 6. I also ran it with `--allow-chrome-as-root` successfully. This seems to another Travis glitch.

**Comment by m90** [comment]
I heard they exist....

Let's keep this open and check if it magically disappears in a few days.

**Comment by m90** [comment]
As it looks like magic unfortunately does not fix this for us, I continued looking into it and actually found the culprit: It breaks when using `puppeteer>=1.4.0`. [Locking it down](https://github.com/mantoni/mochify.js/commit/abf1e0667356388d2322493f1d8169b4c06504ff) to `1.3.0` (which the last green build was using)  makes the build green again.

I don't really know what to make of this yet, but I guess I should find some time tomorrow to check the 1.4.0 changelog and see if there are hints in there. Let me know if you have an ideas.

**Comment by mantoni** [comment]
Thanks. I'm tempted to simply skip those few failing tests on Node 6. It's only on Travis and doesn't happen locally, so a simple `if` on some Travis environment variable would do the trick. I don't really want to lock the Puppeteer dependency because of some Travis voodoo. What do you think?

**Comment by m90** [comment]
I'd definitely also **not** lock down Puppeteer just to silence Travis, but I'd still be interested if there are ways that this might break real world test suites.

I've got an entire train ride at hand tomorrow to further poke around this, so I think I'll try to find out what's going on a little more. If I cannot find anything, I'll simply open a PR for skipping the https-es on Node <= 8.

**Comment by mantoni** [comment]
Awesome! I'm not going to stop you 😄

**Comment by m90** [comment]
So this starts getting interesting:

when you lock the puppeteer version to `1.4.0`, it will fail consistently, both on [Travis across all versions](https://travis-ci.org/mantoni/mochify.js/builds/380080401) and for me locally.

I think the underlying root cause for what we've seen until now is the following: when running on Travis, Node 6 and 7 use npm3 which doesn't know about `package-lock.json` yet, which is why it installs `1.4.0`, newer Node versions will run npm5 and therefore respect the lock file which means that `1.3.0` is installed because that's what the [lockfile says](https://github.com/mantoni/mochify.js/blob/0c35a32d265bddeb3627842e82b87926bfb50858/package-lock.json#L3204).

This means it's probably not only a Travis thing :disappointed: My gut feeling is that the bump in the Chromium version used is what's breaking things (and it stopped liking the cert), but I have the entire day to find out what's going on.

One question though: I assume the certificate in `fixture` is a self-signed one that you created on your machine?

**Comment by m90** [comment]
Closed via #173
